lib/katello/tasks/clean_candlepin_orphaned_products.rake

```diff
@@ -0,0 +1,41 @@
+require File.expand_path("../engine", File.dirname(__FILE__))
+
+namespace :katello do
```

Comment:
Would it make sense to invoke this rake task from inside your upgrade task? It doesn't seem mandatory given your other changes but it might be useful. You'll know that any orphaned products that occur will have happened after this code is running which might help you narrow down the root cause of where they come from. ie you wouldn't have to worry about sifting through orphaned products that could be half a decade old. 🤔

Reply:
We were actually talking a bit about that before.. Now that you bring up that scenario, it sounds like a good idea to me!